### PR TITLE
CSV Integration for On-demand Reports

### PR DESCRIPTION
**Pull request by davidcui-amzn**
*Issue #, if available:*
N/A
*Description of changes:*
Integrate CSV with client-side for on-demand report generation
* Ability to download CSV from saved searches for On-demand report definitions
* Ability to download CSV reports on-demand from the reporting landing page 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: kibana-reports/server/routes/utils/dataReportHelpers.ts [hunk 1]
```diff
@@ -186,10 +186,10 @@ export const getEsData = (arrayHits, report, params) => {
 export const convertToCSV = async (dataset) => {
   let convertedData: any = [];
   const options = {
-    delimiter: { field: ',' },
+    delimiter: { field: ',', eol: '\n' },
     emptyFieldValue: ' ',
   };
-  await converter.json2csvAsync(dataset, options).then((csv) => {
+  await converter.json2csvAsync(dataset[0], options).then((csv) => {
```
Comment (dai-chen) [inline kibana-reports/server/routes/utils/dataReportHelpers.ts:192]: Thanks for the fix. Do you happen to know what caused this? Earlier I tested by UT and manual IT but didn't see this issue.

If necessary, please change mock data accordingly in https://github.com/opendistro-for-elasticsearch/kibana-reports/blob/dev/kibana-reports/server/routes/utils/__tests__/savedSearchReportHelper.test.ts#L371.

Comment (davidcui-amzn) [reply kibana-reports/server/routes/utils/dataReportHelpers.ts:192]: This line changes the format of the data string to be parsed correctly into the CSV file on the client-side. The issue was in the `sanitize()` method in `dataReportHelpers`, it threw the error when a non-string field tried to be parsed with `startsWith` (e.g AvgTicketPrice: 143.83 or something similar).

**Review by zhongnansu — APPROVED** [review]

File: kibana-reports/server/routes/utils/dataReportHelpers.ts (hunk 1 above)
Comment (dai-chen) [reply kibana-reports/server/routes/utils/dataReportHelpers.ts:192]: I see. You're talking about the changes in `sanitize()` right? Curious why `dataset` is changed to `dataset[0]`? Does the UT still pass?

Comment (davidcui-amzn) [reply kibana-reports/server/routes/utils/dataReportHelpers.ts:192]: Yeah, talking about the changes in sanitize(). Changing `dataset` to `dataset[0]` allowed the data string to be parsed into the CSV file correctly when exporting- need to modify the UTs

**Review by dai-chen — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**